src/app/prevention/strides-against/strides-against.component.ts

```diff
@@ -0,0 +1,231 @@
+import { DashboardService } from '../../services/dashboard.service';
+
+
+@Component({
+  selector: 'app-strides-against',
+  templateUrl: './strides-against.component.html',
+  styleUrls: ['./strides-against.component.scss']
+})
+export class StridesAgainstComponent implements OnInit {
+  @ViewChild('gameArea') gameArea;
+  paper: RaphaelPaper;
+  man: RaphaelElement;
+
+  language: any;
+  alerts: any;
+  x = 0;
+  pos = 0;
+  question: string;
+  result: any;
+
+  completed = false;
+  activityComplete = false;
+
+  elements: RaphaelElement[] = [];
+
+  answer: boolean;
+
+
+  BackgroundConfig = BackgroundConfig;
+  ObjectsConfig = ObjectsConfig;
+  ManConfig = ManConfig;
+
+
+  constructor(public languageService: LanguageService, public dialog: MatDialog, public sharedDataService: SharedDataService,
+              public router: Router, public dashboardService: DashboardService) {
+  }
+
+  handleQuestion(): void {
+    const index = this.pos - 1;
+    this.elements[index].hide();
+    this.question = this.ObjectsConfig[index].element.question;
+    const answer = this.ObjectsConfig[index].element.answer;
+    const image = this.ObjectsConfig[index].element.messageImageURL;
+    const message = this.ObjectsConfig[index].element.messageBody;
+    const messageSuccessBody = '<img src = "' + image + '" width=auto height=200px><br><br>' + message;
+    const messageFailBody = '<img src = "' + image + '" width=auto height=200px><br><br>' + message + '' +
+      '<br><br><a href="/"> Go to Dashboard </a>';
+
+    this.openDialog().subscribe(res => {
+      if (this.result === answer) {
+        swal('That\'s correct.', messageSuccessBody, 'success');
+      } else {
+        this.man.animate({transform: 'r90'}, 2000, 'linear', () => {
+          swal({
+            type: 'error',
+            title: 'Incorrect',
+            html: messageFailBody,
+            confirmButtonText: 'Try again',
+          }).then(() => location.reload());
+          this.man.toBack();
+        })
+      }
+    });
+  }
+
+  openDialog(): Observable<boolean> {
+    const dialogRef = this.dialog.open(DialogComponent, {
+      data: {
+        question: this.question,
+      },
+      position: {
+        top: '300px',
+        left: '560px'
+      },
+      disableClose: true,
+      autoFocus: true,
+      width: '482px'
+    });
+    dialogRef.afterClosed().subscribe(res => this.result = res);
+    return dialogRef.afterClosed();
+  }
+
+  checkXMoveSign(position: number): number {
+    if (position < 8 || (position < 24 && position > 17)) {
+      return 1
+    }
+    if ((position > 9 && position < 16) || (position <= 32 && position > 25)) {
+      return -1
+    }
+  }
+
+  moveMan(current: number, final: number, yMove: boolean): void {
+    if (yMove) {
+      switch (Math.floor(final / 8)) {
+        case 1: {
+          const x = (8 - current) * 86;
+          this.man.animate({transform: '...t' + x + ',0'}, 1000, 'linear', () => {
```

Comment:
Lot of code repetition here.. Move it to a function and reuse it..

Reply:
Done.